packages/calcite-components/src/components/slider/slider.e2e.ts

```diff
@@ -108,1219 +112,1351 @@ describe("calcite-slider", () => {
+        expect(await element.getProperty("maxValue")).toBe(minMaxValue);
+        expect(changeEvent).toHaveReceivedEventTimes(0);
+        expect(inputEvent).toHaveReceivedEventTimes(0);
+      }
 
-    describe("at min edge", () => {
-      const expectedValue = 5;
+      async function setUpTest(sliderAttrs: string): Promise<void> {
+        page = await newE2EPage();
+        await page.setContent(html`<calcite-slider ${sliderAttrs}></calcite-slider>`);
 
-      it("click/tap should grab the max value thumb", async () => {
-        await setUpTest(`${commonSliderAttrs} min-value="${expectedValue}" max-value="${expectedValue}"`);
+        element = await page.find("calcite-slider");
+        trackRect = await getElementRect(page, "calcite-slider", `.${CSS.track}`);
+        changeEvent = await element.spyOnEvent("calciteSliderChange");
+        inputEvent = await element.spyOnEvent("calciteSliderInput");
+      }
 
-        await assertValuesUnchanged(5);
+      describe("at min edge", () => {
+        const expectedValue = 5;
 
-        await page.mouse.click(trackRect.x, trackRect.y);
-        await page.waitForChanges();
+        it("click/tap should grab the max value thumb", async () => {
+          await setUpTest(
+            `${commonSliderAttrs} min-value="${expectedValue}" max-value="${expectedValue}" layout=${layout}`,
+          );
 
-        const isMaxThumbFocused = await isElementFocused(page, `.${CSS.thumbValue}`, { shadowed: true });
+          await assertValuesUnchanged(5);
 
-        expect(isMaxThumbFocused).toBe(true);
-        await assertValuesUnchanged(5);
-      });
+          layout === "horizontal"
+            ? await page.mouse.click(trackRect.x, trackRect.y)
+            : await page.mouse.click(trackRect.x, trackRect.y + trackRect.height);
+          await page.waitForChanges();
 
-      it("mirrored: click/tap should grab the max value thumb", async () => {
-        await setUpTest(`${commonSliderAttrs} min-value="${expectedValue}" max-value="${expectedValue}" mirrored`);
+          const isMaxThumbFocused = await isElementFocused(page, `.${CSS.thumbValue}`, { shadowed: true });
 
-        await assertValuesUnchanged(5);
+          expect(isMaxThumbFocused).toBe(true);
+          await assertValuesUnchanged(5);
+        });
 
-        await page.mouse.click(trackRect.x + trackRect.width, trackRect.y);
-        await page.waitForChanges();
+        it("mirrored: click/tap should grab the max value thumb", async () => {
+          await setUpTest(
+            `${commonSliderAttrs} min-value="${expectedValue}" max-value="${expectedValue}" mirrored layout=${layout}`,
+          );
 
-        const isMaxThumbFocused = await isElementFocused(page, `.${CSS.thumbValue}`, { shadowed: true });
+          await assertValuesUnchanged(5);
 
-        expect(isMaxThumbFocused).toBe(true);
-        await assertValuesUnchanged(5);
+          layout === "horizontal"
```

Comment:
Can you DRY this up by storing the test x, y variables based on layout and passing them to `click`? Applies to similar test steps.